src/middleware/withAPIPageLogger.ts

```diff
@@ -0,0 +1,56 @@
+import { createLogger } from "@fewlines/fwl-logging";
+import { NextApiRequest, NextApiResponse } from "next";
+import { Handler } from "src/pages/api/identity";
+
+type WithAPIPageLogger = (
+  request: NextApiRequest,
+  response: NextApiResponse,
+) => Promise<JSON | void>;
+
+export function withAPIPageLogger(handler: Handler): WithAPIPageLogger {
+  const logger = createLogger("connect-account", "json").withMeta({
+    process: "getServerSideProps",
+  });
+
+  const processTimeStart = process.hrtime.bigint();
+
+  return async (request, response) => {
+    handler(request, response)
+      .then((data) => {
+        const processTimeEnd = process.hrtime.bigint();
+
+        logger.log("Success", {
+          method: request.method,
+          statusCode: request.statusCode,
+          path: request.url,
+          duration: (
+            (processTimeEnd - processTimeStart) /
+            BigInt(1000)
+          ).toString(),
+        });
+
+        response.statusCode = 200;
```

Comment:
With this method, we can't send anything other than 200, which could pose a problem in the future

Reply:
Yeah, I went with the simplest approach as I'm still new to the `statusCode` handling game, and as I don't know the "good" implementation, I just went for this